src/main/ruby/core/thread.rb

```diff
@@ -434,3 +439,35 @@ def inspect
     to_s.inspect
   end
 end
+
+Truffle::KernelOperations.define_hooked_variable(
+  :$SAFE,
+  -> { Thread.current.safe_level },
+  -> v { v = Truffle::Type.check_safe_level(v)
```

Comment:
Is `v` a parameter? Is it normally written in parentheses? It looks odd on its own like that.

Reply:
That the usual syntax afaik.
It could be named `value` though for clarity.